Makefile

```
@@ -40,11 +40,18 @@ notice:
 		-noticeOut NOTICE.txt \
 		-depsOut ""
 
+
+## generate-k8s: Regenerates all k8s templates.
+.PHONY: generate-k8s
+generate-k8s:
+	@$(MAKE) -C deploy/kubernetes all
+
 ## check-ci: Run all the checks under the ci, this doesn't include the linter which is run via a github action.
 .PHONY: check-ci
 check-ci:
 	@mage update
 	@$(MAKE) notice
+	@$(MAKE) generate-k8s
```

I'd propose to change `all` with `generate-k8s` in https://github.com/elastic/elastic-agent/blob/main/deploy/kubernetes/Makefile and then `generate-k8s:` won't be needed

```suggestion
	@$(MAKE) -C deploy/kubernetes generate-k8s
```